vignettes/stan-help.Rmd

```diff
@@ -0,0 +1,403 @@
+#>                 0, 0.01))
+#>             if (data$model_refp > 1) {
+#>                 init$refp_sd_beta <- array(rnorm(data$refp_fncol, 
+#>                   0, 0.01))
+#>             }
+#>         }
+#>         if (data$refp_rncol > 0) {
+#>             init$refp_mean_beta_sd <- array(abs(rnorm(data$refp_rncol, 
+#>                 priors$refp_mean_beta_sd_p[1], priors$refp_mean_beta_sd_p[2]/10)))
+#>             if (data$model_refp > 1) {
+#>                 init$refp_sd_beta_sd <- array(abs(rnorm(data$refp_rncol, 
+#>                   priors$refp_sd_beta_sd_p[1], priors$refp_sd_beta_sd_p[2]/10)))
+#>             }
+#>         }
+#>         return(init)
+#>     }
+#>     return(fn)
+#> }
+#> <bytecode: 0x55927f15dbb0>
+#> <environment: 0x559280451700>
+#> 
+
+```
+
+In the above example, if you have some reason to believe that the standard deviation of the means used could we smaller, you could reduce them by half to 0.5.
+
+```r
+
+new_priors <- default_priors$priors
+
+new_priors[ ,sd := 0.5]
+
+```
+
+You could then pass these new priors to the `epinowcast` function.
+
+```r
+epinowcast(pobs,
+  expectation = expectation_module,
+  fit = fit,
+  model = multithread_model,
+  priors = new_priors
+)
+```
+Similarly, you could take an emperical Bayes approach and use the posteriors of a [fitted model as described in another vignette](https://package.epinowcast.org/articles/germany-age-stratified-nowcasting.html#using-the-inflated-posterior-as-a-prior).
+
+
+## Exploring your data {.modelData}
+
+<!--
+#TODO Non-stationarity, longer time scales, multi-modality from known etiologies
+-->
+
+It is important to understand the data being passed to `epinowcast` and have some ideas regarding the data generating process for your data.
+For instance, if there are significant changes in your reporting triangle, this could manifest as multi-modality in the posterior distribution (i.e., Stan is trying to fit two different reporting delays with the same parameters).
+These different reporting regimes could result in long model run times, very wide posteriors, and ultimately nonsensical inferences.
+Given the nuances of your data, you may need to identify and fit different regimes of your data.
+These problems can also manifest over longer timescales, so while more data are generally better, too much data can result in ill-fitting models.
+
+
+## Common failure modes
+
+### My model takes too long to run
+
+One of the more common issues with Bayesian inference is that the sampler may take a long time to run.
+This can occur because the sampler is having difficulty exploring the posterior parameter space given the model and your data.
+You can a few simple steps to assist with this:
+
+
+1. Increase the [`adapt_delta`](.adaptDelta) argument to 15 which will allow longer trajectories to used for steps when sampling. This is especially important if the `per_at_max_treedepth` value is high in the returned CmdStan diagnostics.
```

Comment:
I think `max_treedepth` is what's meant here?

Reply:
Sorry I meant `adapt_delta` shouldn't be increased to 15 and I think `max_treedepth` is what's meant instead